### PR TITLE
add custom object

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards https://github.com/giantswarm/giantswarm/issues/1197.

### CODE REVIEW

File: custom_object.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+	"k8s.io/client-go/pkg/api/v1"
+)
+
+// CustomObject represents the Flannel TPR's custom object. It holds the
+// specifications of the resource the Flannel operator is interested in.
+type CustomObject struct {
+	unversioned.TypeMeta `json:",inline"`
+	v1.ObjectMeta        `json:"metadata,omitempty"`
+
+	Spec Spec `json:"spec" yaml:"spec"`
```
Comment (kopiczko) [inline custom_object.go:14]: Is there any reason why there is yaml support? I think it only messes with ugoji. cc @rossf7

File: spec.go [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package flanneltpr
+
+import (
+	"github.com/giantswarm/flanneltpr/host"
+	"github.com/giantswarm/flanneltpr/network"
+)
+
+type Spec struct {
+	Host    host.Host       `json:"host" yaml:"host"`
+	Network network.Network `json:"network" yaml:"network"`
```
Comment (kopiczko) [inline spec.go]: I'm not sure how to obtain cluster ID -- it's needed to observe if namespace is gone.

File: custom_object.go (hunk 1 above)
Comment (xh3b4sd) [reply custom_object.go:14]: This is just me copy/paste. Don't know.

File: spec.go (hunk 2 above)
Comment (xh3b4sd) [reply spec.go]: Good catch. Should I add a namespace property to the spec?

Comment (kopiczko) [reply spec.go]: Or `ClusterName` or `Cluster.Name` or `You.Name.It`.

**Comment by xh3b4sd** [comment]
@kopiczko please have another look.

File: meta.go [hunk 3]
```diff
@@ -4,7 +4,7 @@ package flanneltpr
 
 const (
 	// Name represents the name of the third party resource within Kubernetes.
-	Name = "flannel.network.giantswarm.io"
+	Name = "flannel-network.giantswarm.io"
```
Comment (kopiczko) [inline meta.go:7]: :heart:

**Review by kopiczko — APPROVED** [review]

**Pull request merged**